### ISSUE TITLE
Drag and Drop in development-branch still broken

### ISSUE THREAD

**Issue by SchuBu** [FluidTYPO3/flux#1514]
Hey there,
creating an CE within a grid element is possible without problems. The tx_flux_column and tx_flux_parent column in database is correctly filled.

BUT:
Dragging within a grid element (e.q. from named "col1" into "col2") results in disappearing of the CE. The database columns tx_flux_column and tx_flux_parent are filled with "NULL". So there is still a problem.

I'm using the flux development branch. TYPO3 at Version 8.7.9

Any ideas how to fix it? Here a video of this bug:
https://youtu.be/gDqK5WiA77E

**Comment by SchuBu** [comment]
I've just tested a bit around:
I only can reproduce the problem when I'm using this extensions directly from GitHub (dev branch):
* flux
* fluidpages
* fluidcontent
* vhs

If I take them from the TYPO Extension Repository there is a warning about the fluidpages needed TYPO3 Version (7.6 needed, I'm using 8.7.9) but if you ignore this, it's just working. The Drag & Drop thing also...

**Comment by NamelessCoder** [comment]
Most likely a duplicate of https://github.com/FluidTYPO3/flux/issues/1502, please check and confirm.

**Comment by SchuBu** [comment]
I don't think so ... it is a problem for all columns. every time you move a content element (also from within a gridelement to outside "normal" page). please watch video above. Thx.

**Comment by NamelessCoder** [comment]
Did you check if the solution found in https://github.com/FluidTYPO3/fluidcontent/issues/418 applies to your case?

**Comment by glucka** [comment]
I think this is not related to https://github.com/FluidTYPO3/fluidcontent/issues/418

typo3 8.7.10
latest dev flux
latest dev fluidpage
without fluidcontent

And i have the same problem.

**Comment by SchuBu** [comment]
@NamelessCoder: this solution did not work for me. I think @glucka is right, it is not related to fluidcontent

**Comment by NamelessCoder** [comment]
> "tx_flux_parent, tx_flux_column" exist in TCA

Yes, but are they also in TCA showitem for your specific content type? That's the thing that was solved in the issue I linked to: adding the fields specifically for your (custom?) CType values is required.

It is not related to fluidcontent. But the solution found in that issue applies to Flux as well.

**Comment by glucka** [comment]
Yes they are also in TCA showitem of my custom CType

**Comment by glucka** [comment]
This is the showitem of my custom content element:
--div--;LLL:EXT:core/Resources/Private/Language/Form/locallang_tabs.xlf:general,--palette--;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:palette.general;general,--palette--;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:palette.headers;headers,pi_flexform,--div--;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:tabs.appearance,--palette--;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:palette.frames;frames,--palette--;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:palette.appearanceLinks;appearanceLinks,--div--;LLL:EXT:core/Resources/Private/Language/Form/locallang_tabs.xlf:language,--palette--;;language,--div--;LLL:EXT:core/Resources/Private/Language/Form/locallang_tabs.xlf:access,--palette--;;hidden,--palette--;LLL:EXT:frontend/Resources/Private/Language/locallang_ttc.xlf:palette.access;access,--div--;LLL:EXT:core/Resources/Private/Language/Form/locallang_tabs.xlf:categories,categories,--div--;LLL:EXT:core/Resources/Private/Language/Form/locallang_tabs.xlf:notes,rowDescription,--div--;LLL:EXT:flux/Resources/Private/Language/locallang.xlf:tt_content.tabs.relation,tx_flux_parent,tx_flux_column,tx_flux_children

**Comment by NamelessCoder** [comment]
Sorry guys - the TCA chunk appears correct, so this should not be the mentioned TCA-showitem-related bug. I don't have any more suggestions for the moment so I will have to come back to this when time allows; or hope someone figures it out and creates a pull request in the mean time.

**Comment by michaelscheel** [comment]
If I revert [this](https://github.com/FluidTYPO3/flux/commit/720ced39c2c5a40266113b1e85ba1160ee851176) commit drag & drop works fine again (TYPO3 8.7.9, flux 8.2.1). I would love to fix it and create a PR, but I guess I would break more things than I would repair.

$newColumnNumber is never passed to $this->contentService->moveRecord, therefore the following code block is never reached in FluidTYPO3\Flux\Service\ContentService:

```
// Fake relative to value - we can get the target from a session variable
list ($parent, $column) = $this->getTargetAreaStoredInSession($relativeTo);
$row['tx_flux_parent'] = $parent;
$row['tx_flux_column'] = $column;
$row['colPos'] = static::COLPOS_FLUXCONTENT;
$row['sorting'] = 0;
```

If you provide me a list with all cases that have to be tested I could try to fix it.

**Comment by pkerling** [comment]
@NamelessCoder it's difficult to say without really understanding how you tried to fix #1415 - 720ced39 seems to change a lot of stuff in seemingly unrelated ways and by that completely broke drag&drop and copy/move basically.  `$newColumnNumber` in `moveRecord_firstElementPostProcess` is now not even used any more? Why is it there at all?

**Comment by wazum** [comment]
The bug shown in the video (element disappears when moving) is fixed with my pull request for #1534 
Please test.

**Comment by breakone** [comment]
The problem is not solved for me yet with the newest development branch and TYPO3 8.7.10 & 8.7.12.

If I drag & drop a CE from one grid-column to another, the following parameters are sent:
```
cmd[tt_content][87][move]: 11
data[tt_content][87][colPos]: 19329
data[tt_content][87][sys_language_uid]: 0
```

The content element is on page 11, colPos from the grid-column is 19329 (generated in `FluidTYPO3\Flux\View\PreviewView registerTargetContentAreaInSession`).

In `FluidTYPO3\Flux\Service\ContentService moveRecord`, there is no tx_flux_parent and tx_flux_column set. On lines 177 and 178 `$parentUid` and `$area` are empty.

So it seams to be an issue, if a CE is dropped to the top of a grid-column (see line 166ff). If I move a CE in between to other CEs in a grid-column, this works.

**Comment by NamelessCoder** [comment]
Solved as part of https://github.com/FluidTYPO3/flux/pull/1563, will be included in release 9.0. Please see the PR for more details, among other things about how to use the migration script.
